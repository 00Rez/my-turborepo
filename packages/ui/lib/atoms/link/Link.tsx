"use client";

import { memo } from "react";

export type LinkProps = {
  newTab?: boolean;
  href: string;
} & React.HTMLAttributes<HTMLAnchorElement>;

export const Link = memo(({ children, href, newTab, ...other }: LinkProps) => {
  return (
    <a
      href={href}
      rel={newTab ? "noreferrer" : undefined}
      target={newTab ? "_blank" : undefined}
      {...other}
    >
      {children}
    </a>
  );
});

Link.displayName = "Link";
