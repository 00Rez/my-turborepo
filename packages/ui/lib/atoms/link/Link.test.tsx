import { describe, it, expect } from "vitest";
import { Link } from "./Link";
import { render } from "@testing-library/react";

describe(Link.name, () => {
  it("renders", () => {
    const { container } = render(<Link href="https://example.com">Example Link</Link>);
    expect(container).toMatchSnapshot();
  });
});
