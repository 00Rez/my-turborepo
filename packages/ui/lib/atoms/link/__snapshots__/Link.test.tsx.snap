// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`undefined > renders 1`] = `
<div>
  <a
    href="https://example.com"
  >
    Example Link
  </a>
</div>
`;
